src/pages/build-apps/guides/authentication.md

```diff
@@ -0,0 +1,250 @@
+    });
+  } else if (userSession.isUserSignedIn()) {
+    // Handle case in which user is already authenticated
+  }
+};
+```
+
+The `isSignInPending` method of the `userSession` object is used to detect whether the user needs to handle a pending authentication state upon page load.
+
+The `handlePendingSignIn` method is then used to handle that state, returning a `userData` object with all the data needed to save the user's information into their session.
+
+The authenticated state can later be detected by the `isUserSignedIn` method in case any particular handling is needed then.
+
+~> Note that implementing `handlePendingSignIn` is especially important for supporting authentication within the context of mobile apps.
+
+If the user has indeed confirmed authentication in the context of a popup window, the authenticator will resolve the pending authentication state automatically with the app within the parent window.
+
+It will then trigger the `finished` function provided above, which can be used similarly to save the user's information into their session as retrieved with `userSession.loadUserData()`.
+
+## Key pairs
+
+Authentication with Stacks makes extensive use of public key cryptography generally and ECDSA with the `secp256k1` curve in particular.
+
+The following sections describe the three public-private key pairs used, including how they're generated, where they're used and to whom private keys are disclosed.
+
+### Transit private key
+
+The transit private is an ephemeral key that is used to encrypt secrets that
+need to be passed from the authenticator to the app during the
+authentication process. It is randomly generated by the app at the beginning of
+the authentication response.
+
+The public key that corresponds to the transit private key is stored in a single
+element array in the `public_keys` key of the authentication request token. The
+authenticator encrypts secret data such as the app private key using this
+public key and sends it back to the app when the user signs in to the app. The
+transit private key signs the app authentication request.
+
+### Identity address private key
+
+The identity address private key is derived from the user's keychain phrase and
+is the private key of the Stacks username that the user chooses to use to sign in
+to the app. It is a secret owned by the user and never leaves the user's
+instance of the authenticator.
+
+This private key signs the authentication response token for an app to indicate that the user approves sign in to that app.
+
+### App private key
+
+The app private key is an app-specific private key that is generated from the
+user's identity address private key using the `domain_name` as input. It is
+deterministic in that for a given Stacks username and `domain_name`, the same
+private key is generated each time.
+
+The app private key is securely shared with the app on each authentication, encrypted by the authenticator with the transit public key.
```

Comment:
"Because the transit key is only stored on the client side, this prevents a man-in-the-middle attack where a server or internet provider could potentially snoop on the app private key".

Reply:
Especially since the Stacks Wallet does not require https domains during authentication (not saying you should put this here, just more context).

Reply:
Adding here as such: 

![image](https://user-images.githubusercontent.com/28991/104126045-ae380100-535a-11eb-9452-e5842175a1ef.png)